flink-ml-lib/src/main/java/org/apache/flink/ml/feature/maxabsscaler/MaxAbsScaler.java

```diff
@@ -0,0 +1,201 @@
+                inputData
+                        .transform(
+                                "reduceInEachPartition",
+                                inputData.getType(),
+                                new MaxAbsReduceFunctionOperator())
+                        .transform(
+                                "reduceInFinalPartition",
+                                inputData.getType(),
+                                new MaxAbsReduceFunctionOperator())
+                        .setParallelism(1);
+
+        DataStream<MaxAbsScalerModelData> modelData =
+                maxAbsValues.map(
+                        (MapFunction<Vector, MaxAbsScalerModelData>)
+                                vector -> new MaxAbsScalerModelData((DenseVector) vector));
+
+        MaxAbsScalerModel model =
+                new MaxAbsScalerModel().setModelData(tEnv.fromDataStream(modelData));
+        ReadWriteUtils.updateExistingParams(model, getParamMap());
+        return model;
+    }
+
+    /**
+     * A stream operator to compute the maximum absolute values in each partition of the input
+     * bounded data stream.
+     */
+    private static class MaxAbsReduceFunctionOperator extends AbstractStreamOperator<Vector>
+            implements OneInputStreamOperator<Vector, Vector>, BoundedOneInput {
+        private ListState<DenseVector> maxAbsState;
+        private DenseVector maxAbsVector;
+
+        @Override
+        public void endInput() {
+            if (maxAbsVector != null) {
+                output.collect(new StreamRecord<>(maxAbsVector));
+            }
+        }
+
+        @Override
+        public void processElement(StreamRecord<Vector> streamRecord) {
+            Vector currentValue = streamRecord.getValue();
+            if (currentValue == null) {
+                throw new RuntimeException("Input column data cannot be null.");
+            }
+
+            if (maxAbsVector == null) {
+                int vecSize = currentValue.size();
+                maxAbsVector = new DenseVector(vecSize);
+                if (currentValue instanceof DenseVector) {
+                    double[] values = ((DenseVector) currentValue).values;
+                    for (int i = 0; i < currentValue.size(); ++i) {
+                        maxAbsVector.values[i] = Math.abs(values[i]);
+                    }
+                } else {
+                    int[] indices = ((SparseVector) currentValue).indices;
+                    double[] values = ((SparseVector) currentValue).values;
+
+                    for (int i = 0; i < indices.length; ++i) {
+                        maxAbsVector.values[indices[i]] =
+                                Math.max(maxAbsVector.values[indices[i]], Math.abs(values[i]));
+                    }
+                }
+            } else {
+                Preconditions.checkArgument(
+                        currentValue.size() == maxAbsVector.size(),
+                        "CurrentValue should has same size with maxVector.");
```

Comment:
nit: How about update the error message as: `The training data should all have same dimensions`.